azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -44,17 +58,39 @@ func resourceArmSecurityCenterSubscriptionPricing() *schema.Resource {
 					string(security.Standard),
 				}, false),
 			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
This will need to be `Optional` to avoid the breaking change.  Setting a default value here (of `default` as before, or `VirtualMachines` if that replaces it?) would be appropriate.

Reply:
`default` is not allowed in the new version of the API, `VirtualMachines` replaces it. I guess in this case, I can remove the `StateUpgraders`. The reasons I made it a required parameter are (1) in the rest API is required and it's nice to keep a consistent behaviour and (2) not having an explicit resource type can be misleading (when reviewing the code the reader may think that the pricing tier affects all the resources within the subscription).

Reply:
@jackofallops do you still think that there should be a default value?